File: demo-java/src/main/java/com/playwire/demo_java/ads/fullscreen/rewardedinterstitial/RewardedInterstitialActivity.java
package com.playwire.demo_java.ads.fullscreen.rewardedinterstitial;

import android.os.Bundle;
import android.os.PersistableBundle;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.intergi.playwiresdk.ads.fullscreen.PWFullScreenAd;
import com.intergi.playwiresdk.ads.fullscreen.rewardedinterstitial.PWRewardedInterstitial;
import com.playwire.demo_java.R;
import com.playwire.demo_java.misc.Constant;

public class RewardedInterstitialActivity extends AppCompatActivity {
    PWRewardedInterstitial rewardedInterstitial;
    /// The ad unit name, e.g. 'banner-320x50', 'interstitial-home', 'rewarded-coins', etc.
    String adUnitName;
    TextView statusTextView;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_rewarded_interstitial);

        adUnitName = getIntent().getStringExtra(Constant.adUnitNameKey);
        statusTextView = findViewById(R.id.status_text_view);
        loadRewardedInterstitial();
    }

    void loadRewardedInterstitial() {
        PWFullScreenAd.Listener listener = new PWFullScreenAd.Listener() {
            @Override
            public void onFullScreenAdLoaded(@NonNull PWFullScreenAd pwFullScreenAd) {
                statusTextView.setText(String.format("✅ The rewarded interstitial \"%s\" is loaded.", adUnitName));

                showRewardedInterstitial();
            }

            @Override
            public void onFullScreenAdFailedToLoad(@NonNull PWFullScreenAd pwFullScreenAd) {
                statusTextView.setText(String.format("❌ Failed to load the rewarded interstitial \"%s\".", adUnitName));
            }

            @Override
            public void onFullScreenAdShowedFullScreenContent(@NonNull PWFullScreenAd pwFullScreenAd) {
                statusTextView.setText(String.format("👍 The rewarded interstitial \"%s\" was successfully shown.", adUnitName));
            }

            @Override
            public void onFullScreenAdFailedToShowFullScreenContent(@NonNull PWFullScreenAd pwFullScreenAd) {
                statusTextView.setText(String.format("❌ Failed to show the rewarded interstitial \"%s\".", adUnitName));
            }

            @Override
            public void onFullScreenAdDismissedFullScreenContent(@NonNull PWFullScreenAd pwFullScreenAd) {
            }

            @Override
            public void onFullScreenAdImpression(@NonNull PWFullScreenAd pwFullScreenAd) {
            }

            @Override
            public void onFullScreenAdReward(@NonNull PWFullScreenAd pwFullScreenAd, @NonNull String type, int amount) {
                // Handle a reward regarding your business objectives.
                statusTextView.setText(String.format("🎉 The reward is earned.\n Type: %s \n Amount: %d", type, amount));
            }
        };

        rewardedInterstitial = new PWRewardedInterstitial(this, adUnitName, listener);

        /// Use `new PWLoadParams().withTargeting()` to pass your custom targets to ad request.
        /// PWLoadParams params = new PWLoadParams().withTargeting(new HashMap<String, String>() {{
        ///   put("age", "18-35");
        ///   put("page", "travel");
        /// }});
        /// rewardedInterstitial.load(params);

        rewardedInterstitial.load();
        statusTextView.setText(String.format("⏳ The rewarded interstitial \"%s\" is loading.", adUnitName));
    }

    void showRewardedInterstitial() {
        if (rewardedInterstitial == null || !rewardedInterstitial.isLoaded()) {
            /// Load rewarded interstitial one more time or notify a user about error.
            return;
        }
        rewardedInterstitial.show();
    }
}

File: demo-kotlin/src/main/java/com/playwire/demo_kotlin/ads/fullscreen/rewardedinterstitial/RewardedInterstitialActivity.kt
package com.playwire.demo_kotlin.ads.fullscreen.rewardedinterstitial

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.intergi.playwiresdk.ads.fullscreen.PWFullScreenAd
import com.intergi.playwiresdk.ads.fullscreen.rewardedinterstitial.PWRewardedInterstitial
import com.playwire.demo_kotlin.R
import com.playwire.demo_kotlin.misc.Constant

class RewardedInterstitialActivity: AppCompatActivity() {
    // The ad unit name, e.g. 'banner-320x50', 'interstitial-home', 'rewarded-coins', etc.
    private lateinit var adUnitName: String

    private var rewardedInterstitial: PWRewardedInterstitial? = null

    private lateinit var statusTextView: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_rewarded_interstitial)
        statusTextView = findViewById(R.id.status_text_view)

        adUnitName = intent.getStringExtra(Constant.adUnitNameKey) ?: ""
        loadRewarded()
    }

    private fun loadRewarded() {
        val listener = object : PWFullScreenAd.Listener {
            override fun onFullScreenAdLoaded(ad: PWFullScreenAd) {
                statusTextView.text = "✅ The rewarded interstitial \"$adUnitName\" is loaded."
                showRewardedInterstitial()
            }

            override fun onFullScreenAdFailedToLoad(ad: PWFullScreenAd) {
                statusTextView.text = "❌ Failed to load the rewarded interstitial \"$adUnitName\"."
            }

            override fun onFullScreenAdShowedFullScreenContent(ad: PWFullScreenAd) {
                statusTextView.text = "👍 The rewarded interstitial \"$adUnitName\" was successfully shown."
            }

            override fun onFullScreenAdReward(ad: PWFullScreenAd, type: String, amount: Int) {
                // Handle a reward regarding your business objectives.
                statusTextView.text = "🎉 The reward is earned.\n Type: $type \n Amount: $amount"
            }

            override fun onFullScreenAdFailedToShowFullScreenContent(ad: PWFullScreenAd) {
                statusTextView.text = "❌ Failed to show the rewarded interstitial \"$adUnitName\"."
            }

        }
        
        rewardedInterstitial = PWRewardedInterstitial(this, adUnitName, listener)
        rewardedInterstitial?.load()

        // Use `PWLoadParams().withTargeting()` to pass your custom targets to ad request.
        // val params = PWLoadParams().withTargeting(
        //   mapOf(
        //      "age" to "18-32",
        //      "page" to "travel"
        //   )
        // )
        // rewardedInterstitial?.load(params)

        statusTextView.text = "⏳ The rewarded interstitial \"$adUnitName\" is loading."
    }

    private fun showRewardedInterstitial() {
        val rewardedInterstitial = rewardedInterstitial ?: return
        if (rewardedInterstitial.isLoaded) {
            rewardedInterstitial.show()
        } else {
            // Load rewarded interstitial one more time or notify a user about error.
        }
    }

}